werkzeug/datastructures.py

```diff
@@ -1153,9 +1153,15 @@ def add(self, _key, _value, **kw):
         The keyword argument dumping uses :func:`dump_options_header`
         behind the scenes.
 
+        In some cases, it would be needed to ensure the order of parameters.
+        For those cases, a list of ``(key, value)`` tuples can be passed in
+        *args.
+
         .. versionadded:: 0.4.1
             keyword arguments were added for :mod:`wsgiref` compatibility.
         """
+        if args:
+            _value = dump_options_header(_value, args)
```

Comment:
I don't think `_` should be replaced with`-` in the `*args` case. If I'm not wrong, that's done so parameters can be added like `add(..., whatever_word=value)`. Since args use key, value tuples which are already strings, users can write the correct parameter name already (and in my opinion, they should be encouraged to do so)